src/observables/CallbackObservable.ts

```diff
@@ -0,0 +1,110 @@
+import {Observable} from '../Observable';
+import {Subscriber} from '../Subscriber';
+import {Scheduler} from '../Scheduler';
+import {Subscription} from '../Subscription';
+import {immediate} from '../schedulers/immediate';
+import {tryCatch} from '../util/tryCatch';
+import {errorObject} from '../util/errorObject';
+
+export class CallbackObservable<T> extends Observable<T> {
+
+  _isScalar: boolean = false;
+  value: T | T[];
+
+  static create<T>(callbackFunc: Function,
+                   ctx: Object = undefined,
+                   selector: Function = undefined,
+                   scheduler: Scheduler = immediate): Function {
+    return (...args): Observable<T> => {
+      return new CallbackObservable(callbackFunc, ctx, selector, args, scheduler);
+    };
+  }
+
+  constructor(private callbackFunc: Function,
+              private ctx,
+              private selector,
+              private args: any[],
+              public scheduler: Scheduler = immediate) {
+    super();
+  }
+
+  _subscribe(subscriber: Subscriber<T | T[]>) {
+    const callbackFunc = this.callbackFunc;
+    const ctx = this.ctx;
+    const selector = this.selector;
+    const args = this.args;
+    const scheduler = this.scheduler;
+
+    let handler;
+
+    if (scheduler === immediate) {
+      if (this._isScalar) {
+        subscriber.next(this.value);
+        subscriber.complete();
+      } else {
+        handler = (...innerArgs) => {
+          let results;
+
+          this._isScalar = true;
+          this.value = innerArgs;
```

Comment:
The `_isScaler` and `value` properties are not necessary. I suspect this is just cruft from working off of the PromiseObservable. Since promises resolve once, and then cache the value, `_isScalar` and `value` were being set to enable PromiseObservables to fast-track in some operators. This can't be true for CallbackObservable, so you can remove all references to `_isScalar` and `value`.